src/App.tsx

```diff
@@ -1,8 +1,70 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
+import TodoList from './components/TodoList';
+import { PreparedTodo, Todo } from './components/types';
+import { todosWithUsers } from './components/api'
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+
+const App: React.FC = () => {
+  const [isLoaded, setLoaded] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [todos, setTodos] = useState<PreparedTodo[]>([]);
+  const [sortedTodos, setSortedTodos] = useState<PreparedTodo[]>([]);
+
+  const handleButtonClick = async () => {
+    setIsLoading(true);
+
+    await todosWithUsers().then(data => setTodos(data));
```

Comment:
what means todosWithUsers?

```suggestion
    await loadTodosWithUsers().then(data => setTodos(data));
```